fsharp-backend/tests/Tests/Execution.Tests.fs

```diff
@@ -348,48 +348,60 @@ let testLambdaPreview : Test =
         Some(AT.NonExecutedResult(DStr "body")))) ]
 
 
-
+/// Test the results that are returned when we're "previewing" (i.e. Analysis)
+///
+/// "Preview" evals involve more tracing, including of unmatched patterns. In
+/// this test, we set up a `match` expr with many patterns and RHSs. Each test
+/// supplies an `expr` to match against, and a list of expected traces to look
+/// for - up to one of the traces will be an Executed result, and the others
+/// should all be NonExecutedResults.
 let testMatchPreview : Test =
-  testTask "test match evaluation" {
-    let mid = gid ()
-    let pIntId = gid ()
-    let pFloatId = gid ()
-    let pBoolId = gid ()
-    let pStrId = gid ()
-    let pNullId = gid ()
-    let pBlankId = gid ()
-    let pOkVarOkId = gid ()
-    let pOkVarVarId = gid ()
-    let pOkBlankOkId = gid ()
-    let pOkBlankBlankId = gid ()
-    let pNothingId = gid ()
-    let pVarId = gid ()
-    let binopFnValId = gid ()
-    let intRhsId = gid ()
-    let floatRhsId = gid ()
-    let boolRhsId = gid ()
-    let strRhsId = gid ()
-    let nullRhsId = gid ()
-    let blankRhsId = gid ()
-    let okVarRhsId = gid ()
-    let okBlankRhsId = gid ()
-    let nothingRhsId = gid ()
-    let okVarRhsVarId = gid ()
-    let okVarRhsStrId = gid ()
-    let varRhsId = gid ()
-
-    let astFor (arg : Expr) =
-      EMatch(
-        mid,
-        arg,
-        [ (PInteger(pIntId, 5L), EInteger(intRhsId, 17L))
+  testList
+    "test match evaluation"
+    [ let matchId = gid ()
```

Comment:
Using a `testList` and the `yield` here is very confusing to me - I don't really understand how it works or why it's necessary. Can you do something simpler, like a `List.map`?

Reply:
Yeah, I get how it's confusing. Struggled with the best thing to do there.

I'm trying to balance a few things:
- making the tests clear
- making each test case a specific `testTask`, rather than all of the checks be in one huge `testTask`
- ensuring test code only gets executed when  relevant tests get run

Specifically, I'm trying to work around the fact that we generate a bunch of `id`s with `gid()`, referencing both in the test-execution code _and_ in the cases. Kinda awkward.

I have a few options:
- how it is now - with the `yield!` hack.
 (negative: confusing)
- go back to how it was
  (negative: can't easily run individual test cases, which I _really_ appreciated with this setup)
- put all of the `gid()` calls _before_ the `testList`
  (negative: "test code" is executed before anything tries to run the test)

I just pushed a commit towards that last option. I don't love it, but maybe it's better? Happy to have input here, but I really don't want to give up the ability to run individual test cases easily. In any case, it doesn't take much computational effort to run `gid()`. WDYT?

Reply:
Yeah this works nicely.